uiowa/uiowa
# Hotfix profiles for sitemap change.

# Problem
The uiowa_profiles sitemap route is current broken:

https://dentistry.uiowa.edu/directory/sitemap.txt
https://education.uiowa.edu/directory/sitemap.txt

## Solution

This changes the `Client::request` method to throw 503s instead of 404s when an error is encountered. It can be misleading when a 404 is returned when there is an actual error. However, I'm ambivalent about this because it looks like requests for /directory/nonexistent-person return a 500 from the profiles API so we can't detect that and still throw 404s for those.

This also changes the sitemap route to `/directory/sitemap` from `/directory/sitemap.txt`. I can't get the `.xml` extension to work so I left it off for now. I suppose we could leave it as `.txt` but that is misleading as it is no longer plain text. No extension gives us more flexibility but does Google need it?

# How to test

- Inspect https://education.dev.drupal.uiowa.edu/directory
  - Make sure the usual profiles functionality works.
-  Also the sitemap loads at https://education.dev.drupal.uiowa.edu/directory/sitemap
- See errors are now thrown instead of 404s for URLs like https://education.dev.drupal.uiowa.edu/directory/nonexistent-person
